core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -319,16 +319,18 @@ private[spark] class SparkSubmit extends Logging {
         // For example we might use the dependencies for downloading
         // files from a Hadoop Compatible fs e.g. S3. In this case the user might pass:
         // --packages com.amazonaws:aws-java-sdk:1.7.4:org.apache.hadoop:hadoop-aws:2.7.6
-        if (isKubernetesClusterModeDriver) {
-          val loader = getSubmitClassLoader(sparkConf)
-          for (jar <- resolvedMavenCoordinates) {
-            addJarToClasspath(jar, loader)
-          }
-        } else if (isKubernetesCluster) {
+        if (isKubernetesCluster) {
           // We need this in K8s cluster mode so that we can upload local deps
           // via the k8s application, like in cluster mode driver
           childClasspath ++= resolvedMavenCoordinates
         } else {
+          if (isKubernetesClusterModeDriver) {
```

Comment:
@ocworld this is not your mistake but this variable name seems to be wrong. As it is in client mode (see its initial value where the `isKubernetesClient` is referenced) and check the documentation: https://spark.apache.org/docs/latest/running-on-kubernetes.html#client-mode. 

@skonto if you agree we can fix this in a separate PR to avoid further confusions.